queryresultio/sparqlxml/src/main/java/org/eclipse/rdf4j/query/resultio/sparqlxml/AbstractSPARQLXMLParser.java

```diff
@@ -278,6 +278,9 @@ protected void reportWarning(String msg) {
 	public Collection<RioSetting<Boolean>> getCompulsoryXmlFeatureSettings() {
 		Set<RioSetting<Boolean>> results = new HashSet<RioSetting<Boolean>>();
 		results.add(XMLParserSettings.SECURE_PROCESSING);
+		results.add(XMLParserSettings.DISALLOW_DOCTYPE_DECL);
+		results.add(XMLParserSettings.EXTERNAL_GENERAL_ENTITIES);
+		results.add(XMLParserSettings.EXTERNAL_PARAMETER_ENTITIES);
```

Comment:
I see this is in line with OWASP: https://www.owasp.org/index.php/XML_External_Entity_(XXE)_Prevention_Cheat_Sheet#XMLReader

Sp that's good.

Reply:
Btw. Would you like to add LOAD_EXTERNAL_DTD too, it's noted in OWASP that it shouldn't be needed but is still used.

A question about naming. You have DISALLOW_DOCTYPE_DECL but not DISALLOW_EXTERNAL_GENERAL_ENTITIES. This feels a bit inconsistent to me.

Reply:
@hmottestad I agree that the naming is inconsisent, but this is how they are named in the Xerces parser features documentation as well (see http://xerces.apache.org/xerces2-j/features.html). I figured it was better to stick to their naming conventions.

As for LOAD_EXTERNAL_DTD, that was already in there, but is implemented as an optional feature, not a compulsory one. I decided not to touch this and only changed its default to `false`.